src/librustc_mir/transform/nrvo.rs

```diff
@@ -0,0 +1,147 @@
+use rustc_index::bit_set::HybridBitSet;
+use rustc_middle::mir::visit::{MutVisitor, PlaceContext};
+use rustc_middle::mir::{self, BasicBlock, Local, Location};
+use rustc_middle::ty::TyCtxt;
+
+use crate::transform::{MirPass, MirSource};
+
+pub struct RenameReturnPlace;
+
+/// Renames the local corresponding to the
```

Comment:
you trailed off here

Reply:
@oli-obk This isn't really worth reviewing at this point. I'm fixing it up now.